src/plugins/visualizations/public/types.ts

```diff
@@ -71,11 +71,18 @@ export interface ISavedVis {
 
 export interface VisSavedObject extends SavedObject, ISavedVis {}
 
-export interface VisResponseValue {
+export interface VisRenderValue {
+  title?: string;
   visType: string;
-  visData: object;
-  visConfig: object;
-  params?: object;
+  visData?: object | null;
```

Comment:
One of the existing visualizations expects a null value for vizData when no data is present. Since I didn't want to introduce a breaking change, I kept the expected types the same and instead just typed the interfaces more strongly.